lib/puppet/type/service.rb

```diff
@@ -73,6 +73,18 @@ def retrieve
         provider.enabled?
       end
 
+      # This only makes sense on systemd systems. Static services cannot be enabled
+      # or disabled.
+      def insync?(current)
+        if provider.respond_to?(:cached_enabled?) && provider.cached_enabled? == 'static'
+          if provider.respond_to?(:disable_static)
+           return provider.disable_static
```

Comment:
Would it be better to return true here after the `disable_static` call, or continue returning true in `disable_static` itself?